app/scripts/components/exploration/components/datasets/colorRangeSlider.tsx

```diff
@@ -0,0 +1,270 @@
+          Rescale
+        </label>
+
+        <USWDSTextInput
+          type='number'
+          id='range slider min'
+          name='min'
+          placeholder={minValRef.current}
+          className={`${textInputClasses} ${
+            inputError.min || inputError.largerThanMax
+              ? 'border-secondary-vivid text-secondary-vivid'
+              : ' border-base-light'
+          }`}
+          value={minValRef.current}
+          onChange={(event) => {
+            minValRef.current = event.target.value;
+            const value = Number(event.target.value);
+
+            if (value > maxVal - minMaxBuffer)
+              return setInputError({ ...inputError, largerThanMax: true });
+            if (value < min || value > max) {
+              return setInputError({ ...inputError, min: true });
+            } else {
+              setInputError({
+                ...inputError,
+                min: false,
+                largerThanMax: false
+              });
+              setMinVal(Math.min(value, maxVal - minMaxBuffer));
+            }
+          }}
+        />
+
+        <div className='position-relative container display-flex flex-align-center flex-justify-center padding-x-2'>
+          <input
+            type='range'
+            min={min}
+            max={max}
+            step='.001'
+            value={minVal}
+            onChange={(event) => {
+              const value = Math.min(
+                Number(event.target.value),
+                maxVal - minMaxBuffer
+              );
+              resetErrorOnSlide(value, 'min');
+              setMinVal(value);
+              minValRef.current = value;
+            }}
+            className={`thumb ${thumbPosition} z-index-300`}
+            style={{
+              zIndex: minVal >= max - 10 * minMaxBuffer ? '500' : '300'
+            }}
+          />
+          <input
+            type='range'
+            min={min}
+            step='.001'
+            max={max}
+            value={maxVal}
+            onChange={(event) => {
+              const value = Math.max(
+                Number(event.target.value),
+                minVal + minMaxBuffer
+              );
+              resetErrorOnSlide(value, 'max');
+              setMaxVal(value);
+              maxValRef.current = value;
+            }}
+            className={`thumb ${thumbPosition} z-400`}
+            style={{
+              zIndex: minVal <= max - 10 * minMaxBuffer ? '500' : '400'
+            }}
+          />
+          <div className='slider width-card position-relative height-3 display-flex flex-align-center'>
+            <div className='slider__track radius-md position-absolute height-05 z-100 width-full display-flex flex-align-center' />
+            <div
+              ref={range}
+              className='slider__range display-flex flex-align-center flex-justify-center radius-md position-absolute height-05 z-200'
```

Comment:
Found a small bug, steps to reproduce:

1. Add the Net Primary Production layer
2. Open the rescale pop up and reduce the max from 0.3 to somewhere in the middle (e.g 0.15)
3. Reload the page
4. Open the rescale pop up again
5. Notice that the range track is correctly set, but not the thumb for the max value


<img width="388" alt="Screenshot 2024-10-17 at 12 28 47" src="https://github.com/user-attachments/assets/6df44f45-0b41-498e-8580-26221bf72894">

Reply:
Good catch, there was a null check that was checking against the wrong prop and reseting the value for the range input.